app/controllers/public/events_controller.rb

```diff
@@ -13,10 +13,17 @@ def index
   # GET /public/events/654331ae-1710-42e5-bdf4-65a03a80c614.json
   def show
     if params[:id] =~ /\A[0-9]+\z/
-      @event = Event.find(params[:id])
+      @event = Event
+        .includes(recordings: :conference)
+        .find(params[:id])
     else
-      @event = Event.find_by(guid: params[:id])
+      @event = Event
+        .where(guid: params[:id])
+        .or(Event.where(slug: params[:id]))
+        .includes(recordings: :conference)
```

Comment:
I did some research and indeed there seems to be a difference: `find_by` [is internally implemented using `where.take`](https://github.com/rails/rails/blob/v5.1.6/activerecord/lib/active_record/relation/finder_methods.rb#L78). Using `.take` instead of `.find` gives you any first record whereas `.find` implicitly appends a `ORDER BY id ASC` to the query.

I am going to do a commit to use `.take` as the `ORDER BY` is definitively unnecessary.

Regarding the potential n+1 situation it is the same case as for the other half of this method: There is no n+1, however, one query can be safed using eager loading.